Files/UserControls/LayoutModes/GridViewBrowser.xaml.cs

```diff
@@ -219,12 +234,24 @@ private async void CommitRename(TextBox textBox)
 
         private void EndRename(TextBox textBox)
         {
-            StackPanel parentPanel = textBox.Parent as StackPanel;
-            TextBlock textBlock = parentPanel.Children[0] as TextBlock;
-            textBox.Visibility = Visibility.Collapsed;
-            textBlock.Visibility = Visibility.Visible;
+            if (App.AppSettings.LayoutMode == 2)
+            {
+                Popup popup =
+                    (textBox.Parent) as Popup;
+                TextBlock textBlock =
+                    (popup.Parent as Grid).Children[1] as TextBlock;
+                popup.IsOpen = false;                
+            }
+            else
+            {
+                StackPanel parentPanel = textBox.Parent as StackPanel;
+                TextBlock textBlock = parentPanel.Children[0] as TextBlock;
+                textBox.Visibility = Visibility.Collapsed;
+                textBlock.Visibility = Visibility.Visible;
+            }
+
             textBox.LostFocus -= RenameTextBox_LostFocus;
-            textBox.KeyDown += RenameTextBox_KeyDown;
+            textBox.KeyDown -= RenameTextBox_KeyDown;
```

Comment:
What purpose of this change?

Reply:
To my understanding that was a bug: you are making the textbox invisible so you should unsubscribe from the keydown event.
Could you confirm this?

Reply:
Looks logical for me.